releasenotes/notes/reno-691a6036754487df.yaml

```diff
@@ -0,0 +1,8 @@
+---
+features:
+  - |
+    Added the arguments ``coupling_map`` and ``target`` to :class:`.HighLevelSynthesis`
+    transpiler pass.
+  - |
+    Added the arguments ``coupling_map``, ``target`` and ``qubits`` to :class:`.HighLevelSynthesisPlugin`.
+    This change is backward-compatible.
```

Comment:
I think having some examples here would go a long way for plugin maintainers.

Reply:
I agree. At the moment I don't have any good examples, but I would after the last part of #9250 gets cleaned up, i.e. actually having some plugins that take `coupling_map` and `qubits` into account. And then we would be able to mention/reference these specific plugins in the plugin interface docs. By the way, thanks for reminding to update these docs, that's done in [54bd00b](https://github.com/Qiskit/qiskit-terra/pull/10477/commits/54bd00b9cdb66e93b81d664b8c8ca433cd6d6e34).